python/pyspark/testing/utils.py

```diff
@@ -291,7 +293,11 @@ def assertDataFrameEqual(df: DataFrame, expected: DataFrame, check_row_order: bo
                 error_class="UNSUPPORTED_DATA_TYPE",
                 message_parameters={"data_type": type(df)},
             )
-        elif not isinstance(expected, DataFrame) and not isinstance(expected, ConnectDataFrame):
+        elif (
+            not isinstance(expected, DataFrame)
+            and not isinstance(expected, ConnectDataFrame)
+            and not isinstance(expected, List)
```

Comment:
```suggestion
            and not isinstance(expected, list)
```

List is for type hint.